src/ucp/proto/proto_common.c

```diff
@@ -707,11 +759,24 @@ void ucp_proto_request_bcopy_abort(ucp_request_t *request, ucs_status_t status)
     ucp_request_complete_send(request, status);
 }
 
+void ucp_proto_request_bcopy_clean(ucp_request_t *request)
+{
+    ucs_assert(ucp_datatype_iter_is_begin(&request->send.state.dt_iter));
+    ucp_datatype_iter_cleanup(&request->send.state.dt_iter, UCP_DT_MASK_ALL);
+    request->flags &= ~UCP_REQUEST_FLAG_PROTO_INITIALIZED;
+}
+
 void ucp_proto_request_zcopy_abort(ucp_request_t *request, ucs_status_t status)
 {
     ucp_invoke_uct_completion(&request->send.state.uct_comp, status);
 }
 
+void ucp_proto_request_zcopy_clean(ucp_request_t *request)
+{
+    ucs_assert(ucp_datatype_iter_is_begin(&request->send.state.dt_iter));
+    ucp_proto_request_zcopy_clean_by_mask(request, UCP_DT_MASK_ALL);
```

Comment:
clean is slowpath, can always use UCP_DT_MASK_ALL. no need to separate ucp_proto_request_zcopy_clean_by_mask

Reply:
`ucp_proto_request_zcopy_clean_by_mask` is used in successfull request completion as well

Reply:
rename this func to ucp_proto_request_zcopy_reset